python/cuml/dask/preprocessing/encoders.py

```diff
@@ -0,0 +1,357 @@
+            self._features = X.columns
+            self._encoders = {
+                feature: X[feature].unique().compute()
+                for feature in X.columns
+            }
+        else:
+            self._features = self.categories.columns
+            self._encoders = dict()
+            for feature in self.categories.columns:
+                self._encoders[feature] = self.categories[
+                    feature].unique().compute()
+                if self.handle_unknown == 'error':
+                    if not X[feature].isin(
+                            self._encoders[feature]).all().compute():
+                        msg = ("Found unknown categories in column {0}"
+                               " during fit".format(feature))
+                        raise KeyError(msg)
+
+        self.drop_idx_ = self._compute_drop_idx()
+        self._fitted = True
+        return self
+
+    def fit_transform(self, X, as_futures=False):
+        """
+        Fit OneHotEncoder to X, then transform X.
+        Equivalent to fit(X).transform(X).
+
+        Parameters
+        ----------
+        X : dask_cudf.DataFrame
+            The data to encode.
+        as_futures : bool
+            Whether or not to return the one hot encoded arrays
+            as futures or to aggregate them in one cupy array.
+        Returns
+        -------
+        out : list of futures if as_futures is True else a cupy array
+            Transformed input.
+        """
+        return self.fit(X).transform(X, as_futures=as_futures)
+
+    @staticmethod
+    @with_cupy_rmm
+    def _func_xform(classes, df, dtype, drop_idx_, handle_unknown):
+        nb_categories = sum(len(s) for s in classes.values())
+        ohe = cp.zeros((len(df), nb_categories), dtype=dtype)
+
+        j = 0
+        for i, feature in enumerate(df.columns):
+            # LabelEncoder part
+            y = df[feature]
+            y = y.astype('category')
+            encoded = y.cat.set_categories(classes[feature])._column.codes
+            encoded = Series(encoded)
+            if encoded.has_nulls and handle_unknown == 'error':
+                raise KeyError("Attempted to encode unseen key")
+
+            # OneHotEncoder part
+            col_idx = cp.asarray(encoded.to_gpu_array(fillna='pandas'))
+            col_idx += j
+            idx_to_keep = col_idx > -1
+            row_idx = cp.arange(len(ohe))[idx_to_keep]
+            col_idx = col_idx[idx_to_keep]
+            ohe[row_idx, col_idx] = 1
+
+            if drop_idx_ is not None:
+                drop_idx = drop_idx_[feature]
+                drop_idx += j
+                mask = cp.ones((ohe.shape[1]), dtype=cp.bool)
+                mask[drop_idx] = False
+                ohe = ohe[:, mask]
+                j -= 1  # account for dropped category in current cats number
+
+            j += len(classes[feature])
+        return ohe
+
+    @with_cupy_rmm
+    def transform(self, X, as_futures=False):
```

Comment:
I don't believe the `as_futures` is necessary.

Reply:
Can you clarify? I thought of `as_futures` as a way to chain operations on distributed collections without aggregating the results of each operations when not needed. In this case, `transform(as_futures=True)` will return a `dask.Array`  with each sub array containing one hot encodings for its own samples. `transform(as_futures=False)` would return the full aggregated cupy array.

Reply:
You are right, I got it by looking at other `DelayedTransformMixin` models.